services/app-web/src/common-code/proto/healthPlanFormDataProto/unlockedWithALittleBitOfEverything.test.ts

```diff
@@ -51,3 +68,93 @@ describe('unlockedWithALittleBitOfEverything migration', () => {
         ).toBe('2022-10-21')
     })
 })
+
+describe('0001_rate_id_migration', () => {
+    it('correctly generates rate ids for old contract and rates submissions', () => {
+        //Get old proto
+        const oldProtoBytes = fs.readFileSync(
+            'src/common-code/proto/healthPlanFormDataProto/testData/unlockedWithALittleBitOfEverything-2022-08-19.proto'
+        )
+
+        const oldFormData = toDomain(oldProtoBytes)
+
+        if (oldFormData instanceof Error) {
+            throw oldFormData
+        }
+
+        // Decode proto
+        const oldProto = decodeOrError(oldProtoBytes)
+
+        if (oldProto instanceof Error) {
+            throw oldProto
+        }
+
+        //Run Migration
+        const migratedProto = rateIDMigration(oldProto)
+        const migratedProtoBytes =
+            genproto.mcreviewproto.HealthPlanFormData.encode(
+                migratedProto
+            ).finish()
+
+        const migratedFormData = toDomain(migratedProtoBytes)
+
+        if (migratedFormData instanceof Error) {
+            throw migratedFormData
+        }
+
+        //Rate certification name generation is a part of a different migration, so to test our rate id migration we will
+        // manually generate a rate id.
```

Comment:
oof - so you are doing this so that symmetric test passes? Wouldn't we have to keep adding here for every additional new migration? Let me know, I can make a ticket about this I think we would want a pattern here.

Reply:
Yes, this is so symmetric tests pass, I'm less concerned about rate name here and more that the migration made the data asymmetrical when encoding and decoding. 

Additional migrations may need this depending on which test proto they use. If we decide to use the earliest test proto version then we would have to keep adding this generation. Testing is still kind of WIP here so I think your suggestion of a ticket for establishing a pattern would be great! It could be lumped into a ticket that fleshes out the migration unit testing. Ideally if we could mock the functions in `migrate_protos.ts`.